components/camel-box/camel-box-api/src/main/java/org/apache/camel/component/box/api/BoxFoldersManager.java

```diff
@@ -189,6 +189,43 @@ public BoxFolder createFolder(String parentFolderId, String folderName) {
     }
 
     /**
+     * Create a folder specified by path from parent folder with given <code>parentFolderId</code>,
+     * creating intermediate directories as required.
+     *
+     * @param parentFolderId
+     *            - the id of parent folder.
+     * @param path
+     *            - Sequence of Box folder names from parent folder to returned
+     *            folder.
+     * @return The last folder in path, no fault will be thrown if it already exists.
+     */
+    public BoxFolder createFolder(String parentFolderId, String... path) {
+        try {
+            LOG.debug("Creating folder with path '" + path + "' in parent_folder(id=" + parentFolderId + ")");
```

Comment:
Use `{}` placeholders in the logger instead of `+` concat

Reply:
Also do the validation first, and then the log afterwards

Reply:
Absolutely, I just handled it the same way as was done previously in the same code base, but I'll revise.